app/src/main/java/com/limelight/binding/input/trackpad/Core/TrackpadFramework.java

```diff
@@ -0,0 +1,786 @@
+                sDirection = "BUTTON_LEFT";
+                break;
+            case BUTTON_RIGHT:
+                sDirection = "BUTTON_RIGHT";
+                break;
+            case BUTTON_MIDDLE:
+                sDirection = "BUTTON_MIDDLE";
+                break;
+            case BUTTON_X1:
+                sDirection = "BUTTON_X1";
+                break;
+            case BUTTON_X2:
+                sDirection = "BUTTON_X2";
+                break;
+        }
+        return sDirection;
+    }
+
+
+    private static final short autoMoveWait = 500;
+    private static final short autoMoveInterval = 8;
+    private static final short autoMoveSense = 3;
+
+    private static final short moveToClickThreshold = 200;
+    private static final short clickToDragThreshold = 200;
+    private static final short dragToDoubleClickThreshold = 200;
+
+    private Timer buttonTimer;
+
+    private static short autoMoveSpaceThreshold = 100;
+
+    public TrackpadFramework(TrackpadListener listener)
+    {
+        trackpadListener = listener;
+        String deviceHead = Build.MODEL.substring(0,5);
+
+        Log.i(TAG,"DeviceName : "+ deviceHead);
+
+        if("SM-X9".equals(deviceHead)){ // Tab S8 Ultra
+            maxXThreshold = 1763;
+            minXThreshold = 0;
+            maxYThreshold = 1599;
+            minYThreshold = 528;
+            autoMoveSpaceThreshold = 100;
+        }else if("SM-T9".equals(deviceHead)||"SM-X8".equals(deviceHead)){ // Tab S7-S8 PLUS
+            maxXThreshold = 1559;
+            minXThreshold = 0;
+            maxYThreshold = 1599;
+            minYThreshold = 780;
+            autoMoveSpaceThreshold = 100;
+        }else {
+            Log.i(TAG,"Not listed : Calibration is required to use AutoMove.");
+            maxXThreshold = minXThreshold = maxYThreshold = minYThreshold = autoMoveSpaceThreshold = -1;
+        }
+    }
+
+    public boolean trackpadHandler(MotionEvent motionEvent){
+        pointerManager(motionEvent);
+        switch (pointerMode){
+            case 0:
+                onePointer(motionEvent);
+                break;
+            case 1:
+                doublePointer(motionEvent);
+                break;
+            case 2:
+                triplePointer(motionEvent);
+                break;
+        }
+        return true;
+    }
+
+    @TargetApi(Build.VERSION_CODES.Q)
+    private float[][] axisManager(MotionEvent motionEvent){
+        float[][] raw = new float[2][5];
+        for(int i = 0; i< motionEvent.getPointerCount(); i++){
+            raw[0][i] = motionEvent.getRawY(i)-minXThreshold;
+            raw[1][i] = motionEvent.getRawX(i)-minYThreshold;
+        }
+        return raw;
+    }
+
+
+    private int maxXThreshold;
+    private int minXThreshold;
+    private int maxYThreshold;
+    private int minYThreshold;
+
+    private void trackPadCalibration(MotionEvent motionEvent){
+        float x = motionEvent.getRawY();
+        float y = motionEvent.getRawX();
```

Comment:
What's the reason for using `getRawX()` and `getRawY()` instead of `getX()` and `getY()`?

Reply:
Let's try getX.
I just used getRawX when I first created it.

We plan to use getX as much as possible because there are compatibility issues with versions below Q.